iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/connector/protocol/thrift/sync/IoTDBThriftSyncClientManager.java

```diff
@@ -132,14 +143,42 @@ private void reconstructClient(TEndPoint endPoint) throws IOException {
               endPoint.getPort()),
           e);
     }
+  }
 
+  public void sendHandshakeReq(
+      Pair<IoTDBThriftSyncConnectorClient, Boolean> clientAndStatus, TEndPoint endPoint) {
     try {
-      final TPipeTransferResp resp =
+      HashMap<String, String> params = new HashMap<>();
+      params.put(
+          PipeConstant.HANDSHAKE_KEY_TIME_PRECISION,
+          CommonDescriptor.getInstance().getConfig().getTimestampPrecision());
+      params.put(PipeConstant.HANDSHAKE_KEY_CLUSTER_ID, PipeRuntimeAgent.getClusterId());
+
+      TPipeTransferResp resp =
           clientAndStatus
               .getLeft()
-              .pipeTransfer(
-                  PipeTransferHandshakeReq.toTPipeTransferReq(
-                      CommonDescriptor.getInstance().getConfig().getTimestampPrecision()));
+              .pipeTransfer(PipeTransferHandshakeV2Req.toTPipeTransferReq(params));
+      if (resp.getStatus().getCode() != TSStatusCode.SUCCESS_STATUS.getStatusCode()) {
+        // Retry to handshake by PipeTransferHandshakeV1Req.
+        if (resp.getStatus().getCode() == TSStatusCode.PIPE_TYPE_ERROR.getStatusCode()) {
+          LOGGER.warn(
```

Comment:
I think info here is better for better compatibility use.

Reply:
Got it.